SO-3183: Fix ECL attribute constraint group cardinalities

zoliMolnar opened the pull request.

Commit:
[ecl] Change conceptsWithGroups field to Multimap of id-group key-values

Commit:
[ecl] Create a new property for each group a relationship is present in

apeteri commented on snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ecl/EclExpression.java:
```diff
@@ -122,26 +123,29 @@ public Expression apply(Expression it) {
-							return FluentIterable.from(input).transform(SnomedRelationship::getSourceId).toSet();
+						public Multimap<String, Integer> apply(SnomedRelationships input) {
+							final Multimap<String, SnomedRelationship> relationshipsBySource = Multimaps.index(input, SnomedRelationship::getSourceId);
+							return Multimaps.transformValues(relationshipsBySource, SnomedRelationship::getGroup);
```
`transformValues` returns a `Multimap` view that allows duplication, see the example below:

```
Multimap<String, String> m = ImmutableSetMultimap.of(
    "a", "123",
    "b", "1234",
    "a", "abc", 
    "a", "d");

Multimaps.transformValues(m, String::length) --> {a=[3, 3, 2], b=[4]}
```

To avoid this, wrap the result of `transformValues` in an `ImmutableSetMultimap` before returning it.

Commit:
SO-3183: Add test case for refinementGroupCardinality, wrap multimap...

... into ImmutableSetMultimap to prevent duplications from
transformValues in EclExpression

nagyo commented on snomed/com.b2international.snowowl.snomed.datastore.tests/src/com/b2international/snowowl/snomed/core/ecl/SnomedEclEvaluationRequestTest.java:
```diff
@@ -725,6 +724,25 @@ public void refinementWithGroupConjunction() throws Exception {
 	}
 	
+	@Test
+	public void refinementGroupCardinality() throws Exception {
```
Please choose a better method name, there is no group cardinality in this test case!

nagyo reviewed (changes requested):
Please change that one method name and we are good to go!

Commit:
SO-3183: Change test method name
